XMonad/Util/NamedScratchpad.hs

```diff
@@ -111,6 +120,12 @@ data NamedScratchpad = NS { name   :: String      -- ^ Scratchpad name
                           , hook   :: ManageHook  -- ^ Manage hook called for application window, use it to define the placement. See @nonFloating@, @defaultFloating@ and @customFloating@
                           }
 
+newtype NSPState = NSPState [NamedScratchpad]
```

Comment:
This should be a set, morally, and it could be if we went all-in on the idea of "names determine the scratchpad uniquely" (which, tbh, the current code assumes anyways).  Opinions?

Reply:
For reference, I've now implemented this as a `Map String NamedScratchpad`.  It's conceptually cleaner, although not really any less code,